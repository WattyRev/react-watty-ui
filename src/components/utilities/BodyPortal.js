import ReactDOM from "react-dom";

/**
 * Body portal creates a ReactDOM portal to send the provided content to the end of the <body>
 * element.
 */
const BodyPortal = ({ children }) => {
  return ReactDOM.createPortal(children, document.querySelector("body"));
};

export default BodyPortal;
