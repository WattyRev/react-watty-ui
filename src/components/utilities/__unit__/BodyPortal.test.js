import React from "react";
import { shallow } from "enzyme";
import BodyPortal from "../BodyPortal";

jest.mock("react-dom");

describe("BodyPortal", () => {
  it("renders", () => {
    expect.assertions(1);
    const wrapper = shallow(
      <BodyPortal>
        <div>test</div>
      </BodyPortal>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
