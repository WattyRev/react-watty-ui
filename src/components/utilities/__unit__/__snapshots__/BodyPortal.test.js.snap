// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyPortal renders 1`] = `""`;
